### PR TITLE
Alan moving

### PR DESCRIPTION
**Pull request by alan24**
Still working on update piece test.

### CODE REVIEW

File: app/controllers/pieces_controller.rb [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
 
-  def show
-    @piece = Piece.find(params[:id])
-    @piece.game.pieces
-  end
-
+  # def show
+  #   @game = current_piece.game
+  #   @pieces = current_piece.game.pieces
+  # end
```
Comment (KaoruDev) [inline app/controllers/pieces_controller.rb]: please delete code if it's not being used =)

**Comment by alan24** [comment]
I deleted the comments :)

File: test/controllers/pieces_controller_test.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
 require 'test_helper'
 
 class PiecesControllerTest < ActionController::TestCase
+  test "should get update" do
+    piece = FactoryGirl.create(:piece)
+    put :update, :id => piece.id, :piece => { :x_position => 2, :y_position => 1 }
+    assert_equal 1, piece.y_position
+    # assert_equal 2, piece.x_position
```
Comment (KaoruDev) [inline test/controllers/pieces_controller_test.rb]: one more spot =P

Comment (KaoruDev) [inline test/controllers/pieces_controller_test.rb]: you actually wanna uncomment this, this is relevant

**Comment by alan24** [comment]
This is the part that I am stuck at. I am trying to pass the test, but the piece will not render to the new position of 2. So I commented the code for now.

File: app/controllers/pieces_controller.rb [hunk 3]
```diff
@@ -1,16 +1,22 @@
...
-    @piece.game.pieces
-  end
-
-
+  def update
+    @game = current_piece.game
 
+    x = params[:x_position]
+    y = params[:y_position]
+    current_piece.update_attributes(:x_position => x, :y_position => y)
```
Comment (KaoruDev) [inline app/controllers/pieces_controller.rb]: replace `(:x_position => x, :y_position => y)` with `piece_params` and then you can delete:

```
x = params[:x_position]
y = params[:y_position]
```

File: test/controllers/pieces_controller_test.rb [hunk 4]
```diff
@@ -1,4 +1,12 @@
 require 'test_helper'
 
 class PiecesControllerTest < ActionController::TestCase
+  test "should get update" do
+    piece = FactoryGirl.create(:piece)
+    put :update, :id => piece.id, :piece => { :x_position => 2, :y_position => 1 }
```
Comment (KaoruDev) [inline test/controllers/pieces_controller_test.rb]: right after this, add `piece.reload`

**TLDR** ActiveRecord caches attributes on the ruby model object so that it doesn't have to keep asking the server for details. When we create send the request to update the record, the model ( ruby object ) and record (postgresql row) are out of sync. `reload` method updates the data in the model ( ruby object ).

**Comment by alan24** [comment]
Thanks Kaoru! 
The test I ran passed. 
I also tried trial and error on my pieces controller, so I kept running the test after one edit.

**Comment by alan24** [comment]
I will merge this branch later in the day, so everyone can look at my code and make sure all my code looks good.

File: test/controllers/pieces_controller_test.rb [hunk 5]
```diff
@@ -1,4 +1,13 @@
...
 
 class PiecesControllerTest < ActionController::TestCase
+  test "should get update" do
+    piece = FactoryGirl.create(:piece)
+    put :update, :id => piece.id, :piece => { :x_position => 1, :y_position => 2 }
+    piece.reload
+    assert_equal 1, piece.x_position
+    assert_equal 2, piece.y_position
+    assert_response :found
+    assert_redirected_to game_path(piece)
```
Comment (KaoruDev) [inline test/controllers/pieces_controller_test.rb]: it shouldn't be redirecting to games#show with id of `piece`, it should redirect to `game.id` right?

**Comment by alan24** [comment]
I thought this code works with the id of piece because the piece gets redirected to games#show with the piece id. So the piece still stays on the chess board with the game redirecting to #show.  
That is how I wrote my pieces_controller.rb except I forgot to add (@game) inside game_path.

File: test/controllers/pieces_controller_test.rb [hunk 6]
```diff
@@ -1,4 +1,14 @@
 require 'test_helper'
 
 class PiecesControllerTest < ActionController::TestCase
+  test "should get update" do
+    game = FactoryGirl.create(:game)
```
Comment (KaoruDev) [inline test/controllers/pieces_controller_test.rb]: you don't need this since your `piece` factory already create a game. =)
